src/sysfiles.c

```diff
@@ -1961,6 +1961,11 @@ Int HasAvailableBytes( UInt fid )
     }
 
 #ifdef HAVE_SELECT
+  // All sockets other than raw sockets are always ready
+  if(syBuf[fid].type != raw_socket) {
+      return 1;
+  }
+  else
```

Comment:
Hmmmm... would it perhaps be better to fall through to the call to `SyIsEndOfFile` below? I.e. just insert this line here:
```C
  if (syBuf[fid].type == raw_socket)
```